### ISSUE TITLE
NullPointerException when using InsertSetStep.set(Map) on plain SQL table with plain SQL field names

### ISSUE THREAD

**Issue by harigyan** [jOOQ/jOOQ#11275]
[How to consume java Map<String,Object> to generate insert and update sql queries in JOOQ (without code generation)?](https://stackoverflow.com/questions/65808357/how-to-consume-java-mapstring-object-to-generate-insert-and-update-sql-queries)

```
Map<String, Object> data = new TreeMap<String, Object>();
// set some key->values
data.put("username", "Hari Gyan");
data.put("col1", val1);
data.put("col2", val2);
DSLContext create; // Autowired
create.insertInto(table(name("myTable")).set(data).execute();
```

### Expected behaviour
It is expected to insert the data of map into the table **myTable**

### Actual behaviour
Throwing NullPointerException : 
```
java.lang.NullPointerException: null
	at org.jooq.impl.LazyVal.<init>(LazyVal.java:62) ~[jooq-3.14.4.jar:na]
	at org.jooq.impl.FieldMapsForInsert.addFields(FieldMapsForInsert.java:333) ~[jooq-3.14.4.jar:na]
	at org.jooq.impl.FieldMapsForInsert.set(FieldMapsForInsert.java:366) ~[jooq-3.14.4.jar:na]
	at org.jooq.impl.InsertQueryImpl.addValues(InsertQueryImpl.java:298) ~[jooq-3.14.4.jar:na]
	at org.jooq.impl.InsertImpl.set(InsertImpl.java:734) ~[jooq-3.14.4.jar:na]
	at org.jooq.impl.InsertImpl.set(InsertImpl.java:124) ~[jooq-3.14.4.jar:na]
	at com.mnit.erp.entity.repository.EntityRepositoryImpl.addEntity(EntityRepositoryImpl.java:26) ~[classes/:na]
        ....
        .....
``` 
### Steps to reproduce the problem

- If the problem relates to code generation, please post your code generation configuration
- A complete set of DDL statements can help re-create the setup you're having
- An MCVE can be helpful to provide a complete reproduction case: https://github.com/jOOQ/jOOQ-mcve

### Versions

- jOOQ: 3.14
- Spring Boot : 2.4
- Spring Framework : 5.3
- Java: 1.8
- Database (include vendor): MySql 
- OS: Mac Os Seirra 10.12.5
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks for your report. I can reproduce the NPE, which seems to appear when combining plain SQL jOOQ usage with the `InsertSetStep.set(Map)` method. Will investigate right away.

**Comment by lukaseder** [comment]
The workaround here would be to use:

```java
Map<Field<?>, Object> data = new TreeMap<>();
// set some key->values
data.put(field(name("username")), "Hari Gyan");
data.put(field(name("col1")), val1);
data.put(field(name("col2")), val2);
DSLContext create; // Autowired
create.insertInto(table(name("myTable")).set(data).execute();
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.6 (#11301)
